src/client/Client.cc

```diff
@@ -13881,16 +13848,12 @@ int Client::_unlink(Inode *dir, const char *name, const UserPerm& perm)
 
   InodeRef otherin;
   Inode *in;
-  Dentry *de;
-
-  int res = get_or_create(dir, name, &de);
-  if (res < 0)
-    goto fail;
+  Dentry *de = get_or_create(dir, name);
   req->set_dentry(de);
   req->dentry_drop = CEPH_CAP_FILE_SHARED;
   req->dentry_unless = CEPH_CAP_FILE_EXCL;
 
-  res = _lookup(dir, name, 0, &otherin, perm);
+  int res = _lookup(dir, name, 0, &otherin, perm);
   if (res < 0)
     goto fail;
```

Comment:
Here and in ll_unlink() you don't remove the 'fail' label like you do in other syscalls. Any reason why?

Reply:
Okay, I missed this one. For the others there had the compile errors so I fixed them. Will fix it soon.